rust-runtime/aws-smithy-http/src/event_stream/input.rs

```diff
@@ -48,19 +48,20 @@ where
     }
 }
 
-/// Adapts a `Stream<SmithyMessageType>` to a signed `Stream<Bytes>` by using the provided
-/// message marshaller and signer implementations.
-///
-/// This will yield an `Err(SdkError::ConstructionFailure)` if a message can't be
-/// marshalled into an Event Stream frame, (e.g., if the message payload was too large).
-#[pin_project]
-pub struct MessageStreamAdapter<T, E> {
-    marshaller: Box<dyn MarshallMessage<Input = T> + Send + Sync>,
-    signer: Box<dyn SignMessage + Send + Sync>,
-    #[pin]
-    stream: Pin<Box<dyn Stream<Item = Result<T, BoxError>> + Send>>,
-    end_signal_sent: bool,
-    _phantom: PhantomData<E>,
+pin_project! {
+    /// Adapts a `Stream<SmithyMessageType>` to a signed `Stream<Bytes>` by using the provided
+    /// message marshaller and signer implementations.
+    ///
+    /// This will yield an `Err(SdkError::ConstructionFailure)` if a message can't be
+    /// marshalled into an Event Stream frame, (e.g., if the message payload was too large).
+    pub struct MessageStreamAdapter {
```

Comment:
The `<T, E>` generic args disappeared? Make sure to test this crate with `--all-features` since the event stream parts aren't on by default.